### PR TITLE
[SAAS-8441] Allow the usage of named Dockerfile when building image for local env

### PR DESCRIPTION
**Pull request by siewert-thanh**

### CODE REVIEW

File: ecrDeployment.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const exec = require('@actions/exec');
  * @param environment {"dev" | "beta" | "prod"}
  * @param version {string}
  * @param shouldBuildImage {boolean}
+ * @param shouldUseCustomDockerfile {boolean}
```
Comment (lisa-tw) [inline ecrDeployment.js:12]: Please document the new param in the `action.yml`

File: ecrDeployment.js [hunk 2]
```diff
@@ -37,17 +39,22 @@ function getAddGithubTokenEnvCommand() {
...
   const addGithubTokenCommand = getAddGithubTokenEnvCommand();
   await exec.exec('docker buildx create --use');
   console.log('Pushing to ECR...');
-  await exec.exec(
-    `docker buildx build ${addGithubTokenCommand} -t ${ecrRepo}:${version}-amd64 --platform linux/amd64 . --push`
-  );
-  await exec.exec(
-    `docker buildx build ${addGithubTokenCommand} -t ${ecrRepo}:${version}-arm64 --platform linux/arm64 . --push`
-  );
+  if(shouldUseCustomDockerfile){
```
Comment (lisa-tw) [inline ecrDeployment.js:47]: There is not need to create two separate methods. You can just create a variable like this and then add it to the build command (same as we do with the `addGithubTokenCommand`

`const dockerfileName = shouldUseCustomDockerfile ? ' -f Dockerfile.local' : ''`

File: index.js [hunk 3]
```diff
@@ -47,7 +49,8 @@ async function run() {
             clusterServiceName,
             environment,
             version,
-            shouldBuildImage
+            shouldBuildImage,
+            shouldUseCustomDockerfile
```
Comment (lisa-tw) [inline index.js:53]: No need to pass the variable here as we will only use the custom Dockerfile if we deploy to local env

Comment (siewert-thanh) [reply index.js]: ok i removed it

File: ecrDeployment.js (hunk 2 above)
Comment (siewert-thanh) [reply ecrDeployment.js]: it's a good idea, i implemented it

**Review by lisa-tw — APPROVED** [review]

**Pull request merged**